v2/test/sample_test.py

```diff
@@ -40,7 +45,8 @@ def v2_sample_test(
     gcs_root: 'URI' = 'gs://ml-pipeline-test/v2',
     image_registry: 'URI' = 'gcr.io/ml-pipeline-test',
     kfp_host: 'URI' = 'http://ml-pipeline:8888',
-    samples_config: list = SAMPLES_CONFIG,
+    samples_config: List[Dict] = SAMPLES_CONFIG,
+    kfp_package_path: 'URI' = 'git+https://github.com/kubeflow/pipelines#egg=kfp&subdirectory=sdk/python'
```

Comment:
With this change, we'll lose the ability to test a commit locally, because KFP package in the local branch is not pushed anywhere accessible. How likely (and often) is this going to be a problem? when we make pipeline spec (IR) changes?

Reply:
After some quick investigation, I think it is possible to sign a GCS url, so that it can be used to download the python sdk in the pipeline. We can apply an short expiration period (like 1h) to avoid the url being leaked in the future.

https://stackoverflow.com/questions/54329585/pip-install-from-a-google-cloud-storage-bucket
https://cloud.google.com/storage/docs/access-control/signed-urls

Reply:
or if we make sure gcs python client is available in sample test v2 components in some way. we can download the sdk tar.gz package locally and then `pip install` it.

Just some ideas, I'm not familiar enough with SDK to say whether any of them is feasible.

Reply:
@Bobgy sorry I was not able to successfully make signed urls work with workload identity (and with local auth). Feel free to open an FR to clean up and assign to me. For now, I'm keeping it as is so we can get this in. Thanks!

Reply:
Thanks for the efforts! I'm fine with leaving this here.